src/layouts/GridView.jsx

```diff
@@ -23,12 +22,11 @@ const useStyles = makeStyles(() => ({
 
 export default function GridView() {
   const classes = useStyles();
-  const { category = "home" } = useParams();
   const [cardData, setCardData] = useState([]);
+  const { category = "home" } = useParams();
   const [loading, setLoading] = useState(true);
 
-  const fetchData = () => {
-    setLoading(true);
+  const fetchData = useCallback(() => {
     axiosPrivate
```

Comment:
Are not using loading ?

Reply:
loading is set true ,  no need to call setLoading in the begging of fetchData function